### PR TITLE
add the eth_getProof call

### PR DESCRIPTION
**Pull request by thinkAfCod**
### What does this PR do?
Add RPC call for eth_getProof

### Where should the reviewer start?
Ethereum.java

### Why is it needed?
In Op Rollup service, it is necessary to call the eth_getProof api to obtain the storageHash value.

### CODE REVIEW

**Comment by NickSneo** [comment]
@thinkAfCod Just 1 comment, other than that LGTM

File: integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java [hunk 1]
```diff
@@ -501,6 +503,19 @@ public void testEthGetLogs() throws Exception {
         assertFalse(logs.isEmpty());
     }
 
+    @Disabled // Not available
```
[the remark this answers is not in the document]
Comment (NickSneo) [reply integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java]: Edit: Test enabled, commit pushed

**Review by NickSneo — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (thinkAfCod) [reply integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java:506]: Sorry, I just saw this comment now. 
Previously, this testing method would return some error messages, and I wasn't sure of the reason.
However, when I used the same code elsewhere, I was able to get a normal response. 
So, I disabled this test case.

[the remark this answers is not in the document]
Comment (thinkAfCod) [reply integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java:506]: It seems that the besu node used cannot retrieve information for this address. 
Here is the request log:
```
10:22:16.824 [Test worker] DEBUG org.web3j.protocol.http.HttpService - --> POST http://localhost:8545/
10:22:16.824 [Test worker] DEBUG org.web3j.protocol.http.HttpService - Content-Type: application/json; charset=utf-8
10:22:16.824 [Test worker] DEBUG org.web3j.protocol.http.HttpService - Content-Length: 184
10:22:16.824 [Test worker] DEBUG org.web3j.protocol.http.HttpService - 
10:22:16.825 [Test worker] DEBUG org.web3j.protocol.http.HttpService - {"jsonrpc":"2.0","method":"eth_getProof","params":["0x7F0d15C7FAae65896648C8273B6d7E43f58Fa842",["0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"],"latest"],"id":4}
10:22:16.825 [Test worker] DEBUG org.web3j.protocol.http.HttpService - --> END POST (184-byte body)
10:22:16.832 [Test worker] DEBUG org.web3j.protocol.http.HttpService - <-- 200 OK http://localhost:8545/ (7ms)
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - vary: origin
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - Content-Type: application/json
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - transfer-encoding: chunked
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - 
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - {
  "jsonrpc" : "2.0",
  "id" : 4,
  "error" : {
    "code" : -32000,
    "message" : "Account not found"
  }
}
10:22:16.833 [Test worker] DEBUG org.web3j.protocol.http.HttpService - <-- END HTTP (111-byte body)
```

[the remark this answers is not in the document]
Comment (thinkAfCod) [reply integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java]: I replaced the address parameter with 0x0000000000000000000000000000000000000000, and now it works. The previous test data was from the https://docs-demo.quiknode.pro/ node, but it is clearly not compatible with the normal test network.

[the remark this answers is not in the document]
Comment (NickSneo) [reply integration-tests/src/test/java/org/web3j/protocol/core/CoreIT.java]: Sounds good, thanks!

**Comment by NickSneo** [comment]
Closing https://github.com/web3j/web3j/issues/1910

**Pull request merged**